### PR TITLE
Implemented `truncGutenbergRichterMFD` from slip rate and rigidity

### PR DESCRIPTION
**Pull request by micheles**
Notice that it was necessary to introduce a classmethod `from_slip_rate`.

### CODE REVIEW

File: openquake/hazardlib/mfd/truncated_gr.py [hunk 1]
```diff
@@ -296,3 +296,16 @@ def from_moment(cls, min_mag, max_mag, bin_width, b_val, moment_rate):
...
         self = cls(min_mag, max_mag, bin_width, a_val, b_val)
         return self
+
+    @classmethod
+    def from_slip_rate(cls, min_mag, max_mag, bin_width, b_val,
+                       slip_rate, rigidity, area):
+        """
+        Calls .from_moment with moment = slip_rate * rigidity * area
+        """
+        mm = 1E-3  # conversion meters -> millimiters
```
Comment (mmpagani) [inline openquake/hazardlib/mfd/truncated_gr.py:306]: Here we work with SI hence we also need to convert rigidity and area accordingly
area *= 1e6 # km^2 to m^2
rigidity *= 1e9 # GPa to Pa
... unless we did it before :)

File: openquake/hazardlib/sourceconverter.py [hunk 2]
```diff
@@ -654,10 +654,21 @@ def convert_mfdist(self, node):
...
             elif mfd_node.tag.endswith('truncGutenbergRichterMFD'):
-                return mfd.TruncatedGRMFD(
-                    a_val=mfd_node['aValue'], b_val=mfd_node['bValue'],
-                    min_mag=mfd_node['minMag'], max_mag=mfd_node['maxMag'],
-                    bin_width=self.width_of_mfd_bin)
+                slip_rate = mfd_node.get('slipRate')
+                rigidity = mfd_node.get('rigidity')
+                if slip_rate:
+                    assert rigidity
+                    # instantiate with a NaN area, to be fixed later on
```
Comment (mmpagani) [inline openquake/hazardlib/sourceconverter.py:661]: Why not simply set NaN the MFD?

**Review by mmpagani — APPROVED** [review]
Apart from the two minor comments LGTM

File: openquake/hazardlib/sourceconverter.py (hunk 2 above)
Comment (micheles) [reply openquake/hazardlib/sourceconverter.py:661]: I need to keep track of the parameters .slip_rate and .rigidity to be used a few lines below

File: openquake/hazardlib/mfd/truncated_gr.py (hunk 1 above)
Comment (micheles) [reply openquake/hazardlib/mfd/truncated_gr.py:306]: That's your job to check ;-)

**Pull request merged**